enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,956 @@
+talking to only pods in their namespace/network, it may make sense to set raw_l3dev_accept to 0. This is set to 1 by
+default to allow legacy ping applications to work over VRFs. Furthermore, a user modifying sysctl settings to allow
+applications to listen across all VRFs will be unsupported. Reasons include there can be odd behavior and interactions
+that occur with applications communicating across multiple VRFs, as well as the fact that this would break the native
+network isolation paradigm offered by this feature.
+
+For host network pods to be able to communicate with pod IPs on their user-defined network, the only supported method
+will be for the applications to bind their socket to the VRF device. Many applications will not be able to support this,
+so in the future it makes sense to come up with a better solution. One possibility is to use ebpf in order to intercept
+the socket bind call of an application (that typically will bind to INADDR_ANY) and force it to bind to the VRF device.
+Note, host network pods will still be able to communicate with pods via services that belong to its user-defined network
+without any limitations. See the next section [Service Access](#service-access) for more information.
+
+Keep in mind that if a host network pod runs and does not bind to the VRF device, it will be able to communicate on the
+default VRF. This means the host networked pod will be able to talk to other host network pods. However, due to nftables
+rules in the host however, it will not be able to talk to OVN networked pods via the default cluster network/VRF.
+
+For more information on how VRFs function in Linux and the settings discussed in this section, refer to
+[https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/networking/vrf.rst?h=v6.1](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/networking/vrf.rst?h=v6.1) for
+more details.
+
+##### Service Access
+
+Host networked pods in a user-defined network will be restricted to only accessing services in either:
+1. The cluster default network.
+2. The user-defined network in which the host networked pod's namespace belongs to.
+
+This will be enforced by iptables/nftables rules added that match on the cgroup of the host networked pod. For example:
+
+```
+root@ovn-worker:/# iptables -L -t raw -v
+Chain PREROUTING (policy ACCEPT 60862 packets, 385M bytes)
+ pkts bytes target     prot opt in     out     source               destination         
+
+Chain OUTPUT (policy ACCEPT 36855 packets, 2504K bytes)
+ pkts bytes target     prot opt in     out     source               destination         
+   17  1800 ACCEPT     all  --  any    any     anywhere             10.96.0.1            cgroup /kubelet.slice/kubelet-kubepods.slice/kubelet-kubepods-besteffort.slice/kubelet-kubepods-besteffort-pod992d3b9e_3f85_42e2_9558_9d4273d4236f.slice
```

Comment:
this is for service, right ? (that's what I interpret from having an IP address as the `destination` value)

i.e. we'll have one of these rules for each host networked pod on a namespace with a primary network per service ?... The scale of this kind of scares me. But honestly, I don't have an alternative.

Reply:
that's right, its potentially a lot of rules with nftables. We will have to scale test and come up with guidance for users. Maybe in the future ebpf would be an alternative.